backend/spindle/views.py

```diff
@@ -7,51 +8,81 @@
 import base64
 
 
+class SpindleError(Enum):
+    SPINDLE = "spindle"
+    LATEX = "latex"
+    GENERAL = "general"
+
+
+class SpindleResponse:
+    def __init__(
+        self,
+        tex: Optional[str] = None,
+        pdf: Optional[str] = None,
+        redirect: Optional[str] = None,
+        error: SpindleError = None,
+    ):
+        self.tex = tex
+        self.pdf = pdf
+        self.error = error
+        self.redirect = redirect
+
+    def jsonResponse(self) -> JsonResponse:
```

Comment:
I would suggest `json_response` as a name for this method for consistency

Reply:
Ah yes, silly JavaScript brain :P